ArcGISRuntimeSDKQt_CppSamples/Layers/GroupLayers/GroupLayers.qml

```diff
@@ -78,14 +78,14 @@ Item {
                     CheckBox {
                         id: parentBox
                         text: name
-                        checked: true
+                        checked: layerVisible
                         onCheckedChanged: layerVisible = checked
                     }
 
                     Repeater {
                         model: sampleModel.getGroupLayerListModel(layerVisibilityListView.currentIndex)
-                        delegate: CheckBox {
-                            checked: true
+                        delegate: RadioDelegate {
+                            checked: index == 0
```

Comment:
Here are some notes from some debugging I did:

Group Layer Exclusive Debug

```
groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: layerVisible
```
First layer is selected. Unable to select new option.
-
```
groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: index === 0
```
First layer is selected. Can select new option. Radio buttons work as expected.
-
```
// groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: index === 0
```
First layer is selected. Can select new option. Radio buttons work as expected.
-
```
// groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: layerVisible
```
Last layer is selected. Can select new option. Radio buttons work as expected.
-
```
layer1->setVisible(false);
groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: layerVisible
```
Second layer is selected. Cannot select new option.
-
```
layer1->setVisible(false);
groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: index === 0
```
First layer (layer1) is selected and displayed as normal. Can select new option. Radio buttons work as expected.
-
```
layer1->setVisible(false);
// groupLayer->setVisibilityMode(GroupVisibilityMode::Exclusive);
checked: layerVisible
```
Last layer is selected. Can select new option. Radio buttons work as expected. layer1 displays as normal when selected.

Reply:
After a bit of testing, it seems that ```GroupVisibilityMode::Exclusive``` and ```checked: layerVisible``` are incompatible in some way. When they are both set, the view will load with the first visible layer selected, but I am unable to select a new layer from the radio group. 

If ```GroupVisibilityMode::Exclusive``` is removed (and defaults to ::Independent), the last visible layer is selected and the radio buttons work as expected.

Alternatively if we set ```checked: index === 0``` and enable ```::Exclusive```, the first layer is selected and displayed (regardless of its visible status) and the radio buttons work as expected.

Lastly if we remove ```GroupVisibilityMode``` and set ```checked: index === 0```, the first option is selected and the radio buttons work as expected.

Reply:
When we add ```Component.onCompleted: checked = layerVisible``` to the end of ```RadioDelegate``` everything works as expected. The only item selected is the first visible within the group layer list and we when we select a different layer, it become the only one enabled. The only issue is _why does this work??_

Reply:
Ok, I think there's something going on with how ```onCheckedChanged: layerVisible = checked``` interacts with ```checked: layerVisible```. There's a lot of looping going around and it's too complicated to succinctly describe in a comment. Basically I think the layer we're trying to deselect reselects itself at the end of the call stack.

So at this point we can either use ```checked: index === 0``` (or potentially a user supplied variable instead of 0) or we can use ```onCheckedChanged: layerVisible = true``` because it negates the looping logic and ensures that the second call (the one to enable the new layer) is the dominant one.

Is there an alternative to ```onCheckChanged``` that just listens for an affirmative state change rather than just any state change?

We can either

Reply:
Sounds like a binding loop. We can leave it with index === 0